bcipy/tasks/rsvp/copy_phrase.py

```diff
@@ -44,368 +71,610 @@ class RSVPCopyPhraseTask(Task):
-        self.rsvp.text = trial_complete_message(self.window, self.parameters)
+        self.rsvp.update_task_state(text=self.spelled_text,
+                                    color_list=['white'])
         self.rsvp.draw_static()
         self.window.flip()
 
-        # Give the system time to process
-        core.wait(self.buffer_val)
+        # Setup the new Stimuli
+        self.rsvp.stimuli_inquiry = inquiry_schedule.stimuli[0]
+        if self.parameters['is_txt_stim']:
+            self.rsvp.stimuli_colors = inquiry_schedule.colors[0]
+        self.rsvp.stimuli_timing = inquiry_schedule.durations[0]
+
+        self.wait()
+
+        if self.parameters['show_preview_inquiry']:
+            stim_times, proceed = self.rsvp.preview_inquiry()
+            if proceed:
+                stim_times.extend(self.rsvp.do_inquiry())
+        else:
+            stim_times = self.rsvp.do_inquiry()
+            proceed = True
+
+        return stim_times, proceed
+
+    def show_feedback(self, selection: str, correct: bool = True):
+        """Display the selection as feedback if the 'show_feedback'
+        parameter is configured.
+
+        Parameters
+        ----------
+        - selection : selected stimulus to display
+        - correct : whether or not the correct stim was chosen
+        """
+        if self.parameters['show_feedback']:
+            feedback = VisualFeedback(self.window, self.parameters,
+                                      self.experiment_clock)
+            feedback.administer(
+                selection,
+                message='Selected:',
+                fill_color=self.parameters['feedback_message_color'])
+
+    def check_stop_criteria(self) -> bool:
+        """Returns True if experiment is currently within params and the task
+        should continue.
+        """
+        if self.copy_phrase == self.spelled_text:
+            self.logger.debug('Spelling complete')
+            return False
+
+        if (self.inq_counter + 1) >= self.parameters['max_inq_len']:
+            self.logger.debug('Max tries exceeded: to allow for more tries'
+                              ' adjust the Maximum inquiry Length '
+                              '(max_inq_len) parameter.')
+            return False
+
+        if self.session.total_time_spent >= (self.parameters['max_minutes'] *
+                                             60):
+            self.logger.debug('Max time exceeded. To allow for more time '
+                              'adjust the max_minutes parameter.')
+            return False
+        return True
+
+    def next_target(self) -> str:
+        """Computes the next target letter based on the currently spelled_text.
+        """
+        if self.copy_phrase[0:len(self.spelled_text)] == self.spelled_text:
+            # if correctly spelled so far, get the next letter.
+            return self.copy_phrase[len(self.spelled_text)]
+        return BACKSPACE_CHAR
+
+    def execute(self) -> str:
```

Comment:
This is a nice reorganization 👍 
It'll probably take me a while to step through all the logic, but the high level process is much more clear.